app/controllers/javascripts_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+class JavascriptsController < ApplicationController
+  DISK_CACHE_PATH = "#{Rails.root}/tmp/javascript-cache"
+
+  skip_before_action(
+    :check_xhr,
+    :handle_theme,
+    :preload_json,
+    :redirect_to_login_if_required,
+    :verify_authenticity_token,
+    only: [:show]
+  )
+
+  before_action :is_asset_path, :no_cookies, only: [:show]
+
+  def show
+    raise Discourse::NotFound unless last_modified.present?
+    return render body: nil, status: 304 if not_modified?
+
+    # Security: safe due to route constraint
+    cache_file = "#{DISK_CACHE_PATH}/#{params[:digest]}.js"
+
+    unless File.exist?(cache_file)
+      content = query.pluck(:content).first
```

Comment:
can't content be nil here, then you will try to write nil to a file which in theory will explode?

Reply:
Good point. In theory `content` at this point cannot be `nil` because:

1. we check to make sure the database entry exist at the start of the request
2. DB constraint to make sure `content` cannot be `nil` (it is at least an empty string)

But I agree to be safe I will add in another 404 check here.